addons/test_mail_full/tests/test_portal.py

```diff
@@ -75,3 +77,44 @@ def test_portal_share_comment(self):
 
         self.assertIn('Test', message.body)
         self.assertEqual(message.author_id, self.partner_2)
+
+    def test_portal_message_fetch(self):
+        """Test retrieving chatter messages through the portal controller"""
+        self.authenticate(None, None)
+        message_fetch_url = '/mail/chatter_fetch'
+        payload = {
+            'jsonrpc': '2.0',
+            'method': 'call',
+            'id': 0,
+            'params': {
+                'res_model': 'mail.test.portal',
+                'res_id': self.record_portal.id,
+                'token': self.record_portal.access_token,
+            },
+        }
+        post_headers = {
+            'Content-Type': 'application/json',
+        }
+
+        def get_chatter_message_count():
+            res = self.url_open(url=message_fetch_url, data=json.dumps(payload), headers=post_headers)
+            return res.json().get('result', {}).get('message_count', 0)
+
+        self.assertEqual(get_chatter_message_count(), 0)
+
+        for _ in range(8):
+            self.record_portal.message_post(
+                body='Test',
+                author_id=self.partner_1.id,
+                message_type='comment',
+                subtype_id=self.env.ref('mail.mt_comment').id,
+            )
+
+        self.assertEqual(get_chatter_message_count(), 8)
+
+        # Empty the body of a few messages
+        for i in (2, 5, 6):
+            self.record_portal.message_ids[-i].body = ""
```

Comment:
Any reason to use "-i" and not "i"?

I'm guessing it's because they appear in the reverse order they were created, but does it really matter?